control/statefbk.py

```diff
@@ -91,26 +105,57 @@ def place(A, B, p):
 
     See Also
     --------
-    place_varga, acker
+    acker, place_varga
     """
-    from scipy.signal import place_poles
-
-    # Convert the system inputs to NumPy arrays
-    A_mat = np.array(A)
-    B_mat = np.array(B)
-    if (A_mat.shape[0] != A_mat.shape[1]):
-        raise ControlDimension("A must be a square matrix")
-
-    if (A_mat.shape[0] != B_mat.shape[0]):
-        err_str = "The number of rows of A must equal the number of rows in B"
-        raise ControlDimension(err_str)
-
-    # Convert desired poles to numpy array
-    placed_eigs = np.array(p)
+    if method == "acker":
+        return acker(A, B, p)
+    if method == "varga":
+        try:
+            return place_varga(A, B, p, dtime=dtime, alpha=alpha)
+        except ControlSlycot as error:
+            warnings.warn("[Pole placement] Fallback strategy: using Tits-Yang method from scipy.")
+            return place(A, B, p)
```

Comment:
If this is kept, I suggest putting the method explicitly in the call to `place`.  Otherwise if the default is changed, the warning will not be accurate.